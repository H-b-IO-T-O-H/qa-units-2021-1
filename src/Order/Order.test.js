jest.mock("../utils/getDate");

/* eslint-disable import/first */
import React from "react";
import {getDate} from "../utils/getDate";
import {configure, shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Order from "./Order.js";
import {fakeOrders} from "../data/fakeOrders";

configure({adapter: new Adapter()});

describe("Order.js", () => {
    beforeAll(() => {
        getDate.mockReturnValue("11 марта, чт, 2021 год");
    });

    afterAll(() => {
        getDate.mockClear();
    });


    test("default render", () => {
        const template = shallow(<Order order={fakeOrders[0]}/>);
        expect(template).toMatchSnapshot();
    });

    test("default render mock call", () => {
        shallow(<Order order={fakeOrders[0]}/>)
        expect(getDate).toHaveBeenCalledWith(fakeOrders[0].date);
    });

    test("render with empty props", () => {
        const order = new Order({});
        const result = order.render()
        expect(result).toBeNull();
    });

    test("render with no items", () => {
        const order = {
            date: 1,
            shop: "shop"
        }

        const template = shallow(<Order order={order}/>);
        expect(template).toMatchSnapshot();
    });

    test("null render with empty item", () => {
        const props = {
            order: {
                shop: null,
                date: "",
            }
        };

        const order = new Order(props);
        const result = order.render()
        expect(result).toBeNull();
    });
});
