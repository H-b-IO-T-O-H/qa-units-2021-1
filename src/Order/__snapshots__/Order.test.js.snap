// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js default render 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    />
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order.js render with no items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      shop
    </span>
    <span
      id="date"
    />
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order.js render without props 1`] = `""`;
